### ISSUE TITLE
Add more/configurable runners

### ISSUE THREAD

**Issue by ranweiler** [ranweiler/liar#6]
Support configurable sample sizes and running strategies to obtain samples, such as performing a linear regression on increasing loop counts.

**Comment by smbolton** [comment]
Hi Joe,
I added a timer and simple output code to https://github.com/smbolton/stm32f3-liar-test, so it is now benchmarking for real!
I'm commenting here, because the only real issue I ran into is with the default 1 million iterations per bench. On the F3, compiled in debug mode, a simple function to initialize a [u32; 128] to 0..127 would have taken 5+ hours to bench! (Release mode: 16 seconds.) Adding configurable iterations per sample and configurable sample size would be a great first step, but it would also be nice the be able to tell liar "run this bench for a maximum of X seconds".
Huh, another issue: it looks like black_box is getting optimized away under some situations. I'll look in to that next.
